### DISCUSSION TITLE
[Bug] NU1101 cannot install Docfx

### DISCUSSION THREAD

**Discussion by HeKr09** [dotnet/docfx#9039] [General]
**Describe the bug**
I want to install docfx but I guess I am to stupid to do so.

**To Reproduce**
Run:
`dotnet tool install -g docfx`

```
C:\Users\me.me\AppData\Local\Temp\2aa18af5-d820-43c7-9428-1ca7f3cadd0e\restore.csproj : error NU1101: Unable
 to find package docfx. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages
The tool package could not be restored.
Tool 'docfx' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool
```

**Expected behavior**
An installation.

**Context (please complete the following information):**
- OS: Windows 10.0.19045 Build 19045
- Docfx version: None

**Additional context**
I have never installed docfx and I am working without any project right now.

**Comment by Lulalaby** [comment]
CNR on:
```
Edition	Windows 11 Enterprise Insider Preview
Version	        22H2
Installed on	‎31/‎07/‎2023
OS build	23511.1000
Experience	Windows Feature Experience Pack 1000.23511.1000.0
```

(oops, overseen that they only have offline packages)

**Comment by filzrev** [comment]
> No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages

It seems `nuget.org` feed is not included as package source. 
What would be the result when running the following command?

> dotnet nuget list source

On my windows environment. following message is displayed.
```
Registered Sources:
  1.  nuget.org [Enabled]
      https://api.nuget.org/v3/index.json
  2.  Microsoft Visual Studio Offline Packages [Enabled]
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
```

If `nuget.org` is not included in above command output.  Then nuget packages souce is customized.
Try to find `NuGet.config`  that affects package sources.

**NuGet config file locations**
https://learn.microsoft.com/en-us/nuget/consume-packages/configuring-nuget-behavior#config-file-locations-and-uses

**Comment by KalleOlaviNiemitalo** [comment]
@HeKr09 it looks like you are using only Visual Studio offline packages and not nuget.org.  See [sources command (NuGet CLI)](https://learn.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-sources) for the command to check and configure that.

**Comment by HeKr09** [comment]
@filzrev @KalleOlaviNiemitalo 
THat sounds good to me and my config is empty.
```
<?xml version="1.0"?>
<configuration>
  <packageSources>
  </packageSources>
</configuration>
```
```
dotnet nuget list source
Registered Sources:
  1.  Microsoft Visual Studio Offline Packages [Enabled]
      C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\
```

According to the article I need to run:
`nuget sources Enable -Name "nuget.org"`
This does not work because PowerShell does not recognize the term nuget.

Can you tell me what to put in the config file?

**Reply by HeKr09** [reply]
@KalleOlaviNiemitalo @filzrev 
Is there any chance to get an installation guide from you?

**Reply by filzrev** [reply]
For temporary workaround.
Try to explicitly specify additional NuGet package source when running install command.

> dotnet tool install docfx -g --add-source https://api.nuget.org/v3/index.json

---

By default. NuGet package source is defined at `%AppData%\Nuget\Nuget.Config`.
With following settings.

```
  <packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
    <add key="Microsoft Visual Studio Offline Packages" value="C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\" />
  </packageSources>
```

On your environment. It seems `nuget.org` entry is removed for some reason.  
For example.
- System administrator explicitly remove this entry to restrict `nuget.org` package access.
- Accidently remove `nuget.org` entry when using Visual Studio's setting pages.
- Manually placed `NuGet.config` file at Solution folder or any folder up to the drive root.  
   and redefine package source with `<clear/>` tag.

**Reply by HeKr09** [reply]
Thank you very much!!!! This solved so many problems! :) It worked!
